website/conferences/views.py

```diff
@@ -146,58 +145,111 @@ def add_poster_by_email(conference, message):
     if node_created and user_created:
         signals.osf4m_user_created.send(user, conference=conference, node=node)
 
-
-def _render_conference_node(node, idx, conf):
-    record = OsfStorageFile.objects.filter(node=node).first()
-
-    if not record:
-        download_url = ''
-        download_count = 0
-    else:
-        download_count = record.get_download_count()
-        download_url = node.web_url_for(
-            'addon_view_or_download_file',
-            path=record.path.strip('/'),
-            provider='osfstorage',
-            action='download',
-            _absolute=True,
-        )
-
-    author = node.visible_contributors[0]
-    tags = list(node.tags.filter(system=False).values_list('name', flat=True))
-
-    return {
-        'id': idx,
-        'title': node.title,
-        'nodeUrl': node.url,
-        'author': author.family_name if author.family_name else author.fullname,
-        'authorUrl': author.url,
-        'category': conf.field_names['submission1'] if conf.field_names['submission1'] in tags else conf.field_names['submission2'],
-        'download': download_count,
-        'downloadUrl': download_url,
-        'dateCreated': node.created.isoformat(),
-        'confName': conf.name,
-        'confUrl': web_url_for('conference_results', meeting=conf.endpoint),
-        'tags': ' '.join(tags)
-    }
-
-
 def conference_data(meeting):
     try:
         conf = Conference.objects.get(endpoint__iexact=meeting)
     except Conference.DoesNotExist:
         raise HTTPError(httplib.NOT_FOUND)
 
-    nodes = AbstractNode.objects.filter(tags__id__in=Tag.objects.filter(name__iexact=meeting, system=False).values_list('id', flat=True), is_public=True, is_deleted=False)
-    ret = []
-    for idx, each in enumerate(nodes):
-        # To handle OSF-8864 where projects with no users caused meetings to be unable to resolve
-        try:
-            ret.append(_render_conference_node(each, idx, conf))
-        except IndexError:
-            sentry.log_exception()
-    return ret
+    return conference_submissions_sql(conf)
+
+def conference_submissions_sql(conf):
+    """
+    Serializes all meeting submissions to a conference (returns array of dictionaries)
 
+    :param obj conf: Conference object.
+
+    """
+    submission1_name = conf.field_names['submission1']
+    submission2_name = conf.field_names['submission2']
+    conference_url = web_url_for('conference_results', meeting=conf.endpoint)
+    abstract_node_content_type_id = ContentType.objects.get_for_model(AbstractNode).id
+    osf_user_content_type_id = ContentType.objects.get_for_model(OSFUser).id
+
+    with connection.cursor() as cursor:
+        cursor.execute(
+            """
+            SELECT  json_build_object(
+                    'id', osf_abstractnode.id,
```

Comment:
The current code sets `id` to the index in the list--not the PK of the nodes. I think this is used by Treebeard as a unique ID, so it doesn't really matter what it is. That said, it's not a good idea to expose PKs if it's not necessary.